formatters/critical-network-chains.js

```diff
@@ -0,0 +1,220 @@
+          `    - Longest chain duration (shorter is better): ${longestDuration.toFixed(2)}ms\n` +
+          '    - Chains\n' +
+              '      ' + urlTree.replace(/\n/g, '\n      ') + '\n';
+          return output;
+        };
+
+      case 'html':
+        // Returns a handlebars string to be used by the Report.
+        return html;
+
+      default:
+        throw new Error('Unknown formatter type');
+    }
+  }
+
+  static _getLongestChainLength(info) {
+    return info.reduce((total, item) => {
+      if (item.totalRequests > total) {
+        return item.totalRequests;
+      }
+
+      return total;
+    }, 0);
+  }
+
+  static _getLongestChainDuration(info) {
+    return info.reduce((total, item) => {
+      if (item.totalTimeBetweenBeginAndEnd > total) {
+        return item.totalTimeBetweenBeginAndEnd;
+      }
+
+      return total;
+    }, 0);
+  }
+
+  /**
+   * Refactors the gatherer's request information into a tree.
+   */
+  static _createURLTree(info) {
+    return info.reduce((tree, item) => {
+      let node = tree;
+      item.urls.forEach((itemURL, index, arr) => {
+        if (!node[itemURL]) {
+          const isLastChild = (index === arr.length - 1);
+          node[itemURL] = isLastChild ? item.totalTimeBetweenBeginAndEnd : {};
+        }
+
+        node = node[itemURL];
+      });
+
+      return tree;
+    }, {});
+  }
+
+  /**
+   * Converts the tree into an ASCII tree.
+   */
+  static _createURLTreeOutput(info) {
+    const urlTree = CriticalNetworkChains._createURLTree(info);
+
+    function write(node, depth, treeMarkers) {
+      return Object.keys(node).reduce((output, itemURL, currentIndex, arr) => {
+        // Test if this node has children, and if it's the last child.
+        const hasChildren = (typeof node[itemURL] === 'object') &&
+            Object.keys(node[itemURL]).length > 0;
+        const isLastChild = (currentIndex === arr.length - 1);
+
+        // If the parent is the last child then don't drop the vertical bar.
+        const ancestorTreeMarker = treeMarkers.reduce((markers, marker) => {
+          return markers + (marker ? '┃ ' : '  ');
+        }, '');
+
+        // Copy the tree markers so that we don't change by reference.
+        const newTreeMakers = treeMarkers.slice(0);
+
+        // Add on the new entry.
+        newTreeMakers.push(!isLastChild);
+
+        // Create the appropriate tree marker based on the depth of this
+        // node as well as whether or not it has children and is itself the last child.
+        const treeMarker = ancestorTreeMarker +
+            (isLastChild ? '┗━' : '┣━') +
+            (hasChildren ? '┳' : '━');
+
+        const parsedURL = CriticalNetworkChains.parseURL(itemURL);
+
+        // Return the previous output plus this new node, and recursively write its children.
+        return output + `${treeMarker} ${parsedURL.file} (${parsedURL.hostname})` +
```

Comment:
output can just be another interpolation here, actually pretty sure all of this can be :dancer:  no need for +